File: 2022/day_1.py


def main(infile):
	new_snack_collection = False
	snack_bag = []
	calorie_totals = []

	with open(infile, 'r', encoding='utf-8-sig') as fin:
		for line in fin:
			line = line.strip('\n')
			if line == '':
				collection_total = sum(snack_bag)
				calorie_totals.append(collection_total)
				snack_bag.clear()

			else:
				snack_bag.append(int(line))

	if len(snack_bag) > 1:
		collection_total = sum(snack_bag)
		calorie_totals.append(collection_total)


	calorie_totals.sort(reverse=True)

	most_calories = calorie_totals[0]
	second_most_calories = calorie_totals[1]
	third_most_calories = calorie_totals[2]

	total_of_top_three = sum([most_calories, second_most_calories, third_most_calories])
	print(f'Total from top 3: {total_of_top_three}')

if __name__=="__main__":
	infile = 'calories_list.csv'
	main(infile)

File: 2022/day_1.rb

def main(infile)
	collection_total = []
    calorie_totals = []
    snack_bag = []

	data = File.readlines(infile).map(&:chomp)
	data.each { 
		|line| 
    	if line.eql? ""
    	    puts "break!"
			collection_total = snack_bag.sum
			calorie_totals << collection_total
			puts "snack_bag: #{snack_bag}"
			puts "collection_total: #{collection_total}"
			puts "calorie_totals: #{calorie_totals}"
			snack_bag.clear()

		else
			snack_bag << line.to_i

		end

	}
	if snack_bag.length > 1
		collection_total = snack_bag.sum
		calorie_totals << collection_total
	end

	sorted_calories = calorie_totals.sort()

	most_calories = sorted_calories[-1]
	second_most_calories = sorted_calories[-2]
	third_most_calories = sorted_calories[-3]
	puts "Max number of calories: #{most_calories}"
	total_of_top_three = [most_calories, second_most_calories, third_most_calories].sum
	puts "Total from top 3: #{total_of_top_three}"

end



infile = 'calories_list.csv'
main(infile)
